addons/payment/wizards/payment_link_wizard.py

```diff
@@ -54,13 +54,15 @@ def default_get(self, fields_list):
         default='all',
         required=True,
     )
+    warning_message = fields.Char(compute="_compute_warning_message")
 
-    @api.onchange('amount', 'description')
-    def _onchange_amount(self):
+    @api.depends('amount', 'amount_max', 'description')
+    def _compute_warning_message(self):
```

Comment:
Well, it works, but I'd instead use `warning_message`  as it is, a warning message :) 

May I suggest keeping the `_on_change_amount`, but that'll set a new variable `is_amount_allowed` to ` false`? You can create a generic warning message in the template that is hidden if the amount is allowed :)

Reply:
Indeed, we are somehow mixing warning logic and link validity, which could be two separate concepts in some cases (not the case for now i think, but we could consider having a warning message that still allows the user to copy the link).